src/nvim/message.c

```diff
@@ -615,7 +626,13 @@ static bool emsg_multiline(const char *s, bool multiline)
 
   // Display the error message itself.
   msg_nowait = false;  // Wait for this msg.
-  return msg_attr_keep((char_u *)s, attr, false, multiline);
+  int ret = msg_attr_keep((char_u *)s, attr, false, multiline);
+
+  if (State & CMDLINE) {
+    redrawcmdline();
```

Comment:
Is it necessary to do this immediately, rather than defer it to the next iteration of cmdline mode? What happens if two error messages are displayed in a row?

Reply:
I see 2 cases where the condition can be true, first from a `Cmdline*` autocmd, and second from a timer callback.

In the autocmd case, the redraw will be done by the next iteration so we are doing an additional `redrawcmdline` which will be cleared by the redraw. In the timer case, it is possible that no redraw will be called, so we have to call the redraw here.

Prior to this, the `redrawcmdline` is called by the `K_EVENT` handler, so it was not necessary.

For multiple error messages in a row, the cmdline will be drawn in between each of them. This should be the same as the current behavior.

Reply:
> For multiple error messages in a row, the cmdline will be drawn in between each of them. This should be the same as the current behavior.

Not necessarily. Consider a function invoked by a timer or autocmd. It could call `:echoerr` twice in a row (or an internal error condition could easily `emsg()` twice in a row)  . The cmdline should then only be drawn after both error messages, not between them.